src/d3d9/d3d9_interfaces.h

```diff
@@ -0,0 +1,201 @@
+   * - \c pInfo->sType \e must be \c VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO
+   * - \c pInfo->pNext \e must be \c nullptr or point to a supported
+   *   extension-specific structure (currently none)
+   * - \c pInfo->queueFamilyIndexCount must be the length of the
+   *   \c pInfo->pQueueFamilyIndices array, in \c uint32_t units.
+   * - \c pInfo->pQueueFamilyIndices must point to a pre-allocated
+   *   array of \c uint32_t of size \c pInfo->pQueueFamilyIndices.
+   * 
+   * \note As of now, the sharing mode will always be
+   *       \c VK_SHARING_MODE_EXCLUSIVE and no queue
+   *       family indices will be written to the array.
+   * 
+   * After the call, the structure pointed to by \c pInfo can
+   * be used to create an image with identical properties.
+   * 
+   * If \c pLayout is not \c nullptr, it will receive the
+   * layout that the image will be in after flushing any
+   * outstanding commands on the device.
+   * \param [out] pHandle The image handle
+   * \param [out] pLayout Image layout
+   * \param [out] pInfo Image properties
+   * \returns \c S_OK on success, or \c D3DERR_INVALIDCALL
+   */
+  virtual HRESULT STDMETHODCALLTYPE GetVulkanImageInfo(
+          VkImage*              pHandle,
+          VkImageLayout*        pLayout,
+          VkImageCreateInfo*    pInfo) = 0;
+};
+
+/**
+ * \brief D3D9 device interface for Vulkan interop
+ *
+ * Provides access to the device and instance handles
+ * as well as the queue that is used for rendering.
+ */
+MIDL_INTERFACE("2eaa4b89-0107-4bdb-87f7-0f541c493ce0")
+ID3D9VkInteropDevice : public IUnknown {
+  /**
+   * \brief Queries Vulkan handles used by DXVK
+   * 
+   * \param [out] pInstance The Vulkan instance
+   * \param [out] pPhysDev The physical device
+   * \param [out] pDevide The device handle
+   */
+  virtual void STDMETHODCALLTYPE GetVulkanHandles(
+          VkInstance*           pInstance,
+          VkPhysicalDevice*     pPhysDev,
+          VkDevice*             pDevice) = 0;
+
+  /**
+   * \brief Queries the rendering queue used by DXVK
+   * 
+   * \param [out] pQueue The Vulkan queue handle
+   * \param [out] pQueueIndex Queue index
+   * \param [out] pQueueFamilyIndex Queue family index
+   */
+  virtual void STDMETHODCALLTYPE GetSubmissionQueue(
+          VkQueue*              pQueue,
+          uint32_t*             pQueueIndex,
+          uint32_t*             pQueueFamilyIndex) = 0;
+
+  /**
+   * \brief Transitions a Texture to a given layout
+   * 
+   * Executes an explicit image layout transition on the
+   * D3D device. Note that the image subresources \e must
+   * be transitioned back to its original layout before
+   * using it again from D3D9.
+   * \param [in] pTexture The image to transform
+   * \param [in] pSubresources Subresources to transform
+   * \param [in] OldLayout Current image layout
+   * \param [in] NewLayout Desired image layout
+   */
+  virtual void STDMETHODCALLTYPE TransitionTextureLayout(
```

Comment:
You should add a note clarifying that the caller is expected to synchronize the device.
Same thing for most of the other functions.